### DISCUSSION TITLE
Grant public access to internal data

### DISCUSSION THREAD

**Discussion by timchkor** [Budibase/budibase#1658] [Help] [answered]
Hi, I'm relatively new on Budibase, and familiarizing on great features of it ! 

I am trying to give an 'Public' access at 'Manage access' tab of internal database, but nothing happens after clicking 'Done' button. It reverts the 'Public' access to 'Basic'.. any tip on this?

![image](https://user-images.githubusercontent.com/85532850/121108085-158fa980-c844-11eb-8364-a1af0841f143.png)

![image](https://user-images.githubusercontent.com/85532850/121108325-846d0280-c844-11eb-98b9-73b21d0b5e01.png)

Thanks!

**Comment by aptkingston** [comment, marked as answer]
Hey @timchkor. Looks like it could be a bug! I'll investigate and raise it as an issue if it is. Thanks for reporting.

**Reply by joebudi** [reply]
I've raised this with @shogunpurple . Fix will be released pretty soon .

**Reply by shogunpurple** [reply]
@timchkor the fix is released - if you update your budibase installation the issue should be gone. Thanks for raising.

**Reply by slfhstr** [reply]
Great to see fixes coming through
But Github main page still says still shows versions as 0.9.1
 How do we know when to start/stop containers if the version number is not changing but containers are ?

**Reply by timchkor** [reply]
@shogunpurple  Thanks for the hot fix. But I still see the latest release is 0.9.30 - released 15 hours ago. Any tips for versioning up? 

[https://github.com/Budibase/budibase/releases](url)

**Reply by shogunpurple** [reply]
@timconsidine You can see all the new release tags on the github tags page https://github.com/Budibase/budibase/tags. Since moving away from the electron app, we don't actually need to "publish" releases anymore on github to update the app. It makes sense though so people can see the latest on github and in discord. Will publish the latest and keep this up to date from here forward. 

@timchkor https://docs.budibase.com/getting-started#managing-your-budibase-installation

**Reply by timchkor** [reply]
Thanks! it perfectly works now on my machine. 

@shogunpurple By the way, I am trying to use JS script at Automation tab with typical JS code - which didn't work. Any tip & sample on JS and Bash scripting?
My sample code is as below;

<script type="text/javascript">  
 alert("Hello World!");  
</script>

**Reply by shogunpurple** [reply]
@timchkor great to hear. 

The JS code that gets executed in automations is executed on the server side using NodeJS, so you cannot use DOM APIs like `alert()`there, or do any kind of dynamic UI.

You just write JS code directly inside the automation block. It is very useful for mutating or mapping data.

For example:

```
// combine the result of 2 automation block results
steps.1.outputs.value * steps.2.outputs.value
```

Then you can use the result of that script in the rest of your automation.

**Reply by timchkor** [reply]
@shogunpurple Thanks again. 

Is the output value 'String' only? or can it be other types?

If it's running code on server side-NodeJS, then, can I execution more complex code on it? Such as;
'const connection = mysql.createConnection({.....' or 'const mysql = require("mysql")...'

**Reply by shogunpurple** [reply]
@timchkor you can't run the code in your example there, because the code is run in a sandboxed environment for security reasons, where the only APIs that you have access to are the NodeJS standard library, and `node-fetch` if you want to make API calls. It would be a huge security concern to allow users to set up and run completely arbitrary code against a company server, or our eventual cloud sandbox.

The output value is whatever JS type the last line of the script executes. `Number`, `String` and `Boolean` should all be valid.

**Reply by mjashanks** [reply]
@timchkor we do want to support custom automation steps at some point - i.e. more complex ones like you mention. It would be good if you could give an example of what you would like to do. The example that you gave can potentially be covered by
1. Adding a MySQL data source
2. Running an execute query step

Would love to hear more about what you are trying to do. If so, would you mind creating a new discussion for it, so we can track separately? Thanks :)

**Reply by timchkor** [reply]
@mjashanks  I was trying to export the data to csv directly from the user side after the filters are applied. I will open a new thread. Thanks!
